### PR TITLE
Lab №1 Var 3

### PR DESCRIPTION
**Pull request by EgorGif**
--------------------------------------------------------------------
-------------lab1 var3  & lab2 var3---------------------------------------------
--------------------------------------------------------------------

### CODE REVIEW

File: src/pack/MyFrame.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import java.awt.event.ActionListener;
+
+public class MyFrame extends JComponent {
+
+    private Timer timer;
+    private String stroke;
+    private String bgColor;
+    private String segmentColor;
+    private double angle = 0;
+    private final double ROTATE = Math.PI/60;
```
Comment (gost402) [inline src/pack/MyFrame.java]: static?

File: src/pack/MyFrame.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        });
+        timer.start();
+    }
+
+    @Override
+    protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
+        Segment seg = new Segment(angle,bgColor,stroke,segmentColor);
+        seg.paint(g);
+        angle+=ROTATE;
```
Comment (gost402) [inline src/pack/MyFrame.java]: add spaces before and after `+=`

File: src/pack/MyFrame.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+public class MyFrame extends JComponent {
+
+    private Timer timer;
+    private String stroke;
+    private String bgColor;
+    private String segmentColor;
+    private double angle = 0;
+    private final double ROTATE = Math.PI/60;
+
+    MyFrame(String stroke, String bgColor, String segmentColor,int delay) {
```
Comment (gost402) [inline src/pack/MyFrame.java]: space after `,`

File: src/pack/MyFrame.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                repaint();
+            }
+        });
+        timer.start();
+    }
+
+    @Override
+    protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
+        Segment seg = new Segment(angle,bgColor,stroke,segmentColor);
```
Comment (gost402) [inline src/pack/MyFrame.java]: spaces after `,`

File: src/pack/MyFrame.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        seg.paint(g);
+        angle+=ROTATE;
+    }
+
+    public static void main(String args[]) {
+        JFrame frame = new JFrame("Segment");
+        MyFrame MF = new MyFrame(args[0], args[1], args[2],30);
```
Comment (gost402) [inline src/pack/MyFrame.java]: space

File: src/pack/Segment.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    private Shape segment;
+    private String bgColor;
+    private String stroke;
+    private String segmentColor;
+    private int x1 = 300;
+    private int width = 200;
+    private int height = 10;
+    private int y1 = 250;
+    private double angle;
+
```
Comment (gost402) [inline src/pack/Segment.java]: leave only 1 empty line

File: src/pack/MyFrame.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+                repaint();
+            }
+        });
+        timer.start();
+    }
+
+    @Override
+    protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
+        Segment seg = new Segment(angle, bgColor, stroke, segmentColor);
```
Comment (gost402) [inline src/pack/MyFrame.java]: You create new Segment object on each `repaint` action. It's better to make this belongs to `MyFrame` and then draw only changing `angle` inside Timer action, `paintComponent` shouldn't be responsible for `angle` change

File: src/pack/Segment.java [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    public void paint(Graphics g) {
+        Graphics2D graphics2D = (Graphics2D) g;
+        AffineTransform at = AffineTransform.getRotateInstance(angle, x1, y1+5);
+        graphics2D.setTransform(at);
+        graphics2D.setStroke(new BasicStroke(Float.valueOf(stroke)));
+        graphics2D.setColor(new Color(Integer.valueOf(bgColor,16)));
+        graphics2D.fillRect(this.x1, this.y1, this.width, this.height);
+        graphics2D.setColor(new Color(Integer.valueOf(segmentColor,16)));
+        graphics2D.draw(segment);
+
```
Comment (gost402) [inline src/pack/Segment.java]: Remove this empty line

**Comment by EgorGif** [comment]
segment

File: src/pack/MyFrame.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+                repaint();
+            }
+        });
+        timer.start();
+    }
+
+    @Override
+    protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
+        segment = new Segment(angle, bgColor, stroke, segmentColor);
```
Comment (gost402) [inline src/pack/MyFrame.java]: You don't want to re-create segment on each `repaint` action. You should create segment only once in constructor and then re-asign Segment angle in Timer action

**Comment by EgorGif** [comment]
screen

**Pull request merged**